### PR TITLE
Chore: Refactor secrets plugin unit tests code

### PR DESCRIPTION
**Pull request by mmandrus**
**What this PR does / why we need it**:
Refactors some ugly unit test code I wrote to bypass the "dogsled" linter by just using a struct.

This corrects code in https://github.com/grafana/grafana/pull/53561 which had the 9.2.0 milestone.

### CODE REVIEW

**Review by guicaulada — APPROVED** [review]

File: pkg/services/secrets/kvstore/plugin_fatal_crash_test.go [hunk 1]
```diff
@@ -22,44 +21,42 @@ import (
...
-	require.NotNil(t, svc)
-	err = svc.Set(context.Background(), 0, "datasource", "postgres", "my secret")
+	require.NotNil(t, p.secretsKVStore)
+	err = p.secretsKVStore.Set(context.Background(), 0, "datasource", "postgres", "my secret")
 	require.NoError(t, err)
-	isFatal, err := isPluginStartupErrorFatal(context.Background(), GetNamespacedKVStore(kvstore))
+	isFatal, err := isPluginStartupErrorFatal(context.Background(), GetNamespacedKVStore(p.kvstore))
 	require.NoError(t, err)
 	require.True(t, isFatal)
 }
```
Comment (evictorero) [inline pkg/services/secrets/kvstore/plugin_fatal_crash_test.go:50]: nit: I like to separate the code into logic units (Method calls with assertions) to make it more readable.

**Review by evictorero — APPROVED** [review]
Leave a nit comment for readability. 
Did you consider using `assert` instead of `require`? [This could be useful](https://github.com/grafana/grafana/blob/main/contribute/style-guides/backend.md#assertions)

File: pkg/services/secrets/kvstore/plugin_fatal_crash_test.go (hunk 1 above)
Comment (leandro-deveikis) [reply pkg/services/secrets/kvstore/plugin_fatal_crash_test.go:50]: you could also just separate them with an empty between setup, execution and validation of the test

Comment (mmandrus) [reply pkg/services/secrets/kvstore/plugin_fatal_crash_test.go]: fixed here https://github.com/grafana/grafana/pull/54231/commits/acb3abdec06bd896a75c3280ad299269958129bb

**Review by leandro-deveikis — APPROVED** [review]

**Pull request merged**